### PR TITLE
feat(design-system): Typography

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: packages/ui-library/src/theme/typography.ts [hunk 1]
```diff
@@ -4,156 +4,54 @@ import { colors } from './colors';
...
-    fontFamily: 'Inter',
-    fontSize: '1.6875rem', // 27/16
-    fontStyle: 'normal',
-    fontWeight: 400,
-    lineHeight: 1.18519, // 32/27
+    lineHeight: 1.167, // 28/24
+    letterSpacing: '-0.0417em', // -1/24
+    color: colors[mode].typography.typo1,
   },
   h4Bold: {
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:33]: can we create variants with mui theme for mobile? I think those bolds are bolder on mobile for the h* components

File: packages/ui-library/src/theme/typography.ts [hunk 2]
```diff
@@ -163,14 +61,6 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
     lineHeight: 1.35, // 21.6/16
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:61]: This should be 19.36/16

File: packages/ui-library/src/theme/typography.ts [hunk 3]
```diff
@@ -179,20 +69,15 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
     color: colors[mode].typography.typo2,
   },
-  body1: {
+  bodyBold: {
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:72]: Seems like body, bodyBold, bodySemiBold all inherit from the same `body` which defines the same lineHeight fontStyle, sfontSize and fontFamily as all the rest, we could create that base one, and make all the rest of this extend that one

File: packages/ui-library/src/theme/typography.ts [hunk 4]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
     color: colors[mode].typography.typo3,
   },
-  bodySmallBold: {
+  bodySmallSemibold: {
     fontFamily: 'Inter',
     fontSize: '0.875rem', // 14/16
     fontStyle: 'normal',
-    fontWeight: 700,
+    fontWeight: 600,
     lineHeight: 1.28571, // 18/14
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:94]: this one should be 17/14

File: packages/ui-library/src/theme/typography.ts [hunk 5]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
-  bodySmallSmall: {
+  // Body Extra Small
+  bodyExtraSmall: {
     fontFamily: 'Inter',
     fontSize: '0.75rem', // 12/16
     fontStyle: 'normal',
     fontWeight: 500,
-    lineHeight: 1.16, // 14/12
-    color: colors[mode].typography.typo3,
+    lineHeight: 1, // 12/12
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:111]: this one is 16/12

File: packages/ui-library/src/theme/typography.ts [hunk 6]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
-  body2: {
+  bodyExtraSmallBold: {
     fontFamily: 'Inter',
-    fontSize: '0.875rem', // 14/16
+    fontSize: '0.75rem', // 12/16
     fontStyle: 'normal',
-    fontWeight: 400,
-    lineHeight: 1.14286, // 16/14
+    fontWeight: 700,
+    lineHeight: 1, // 12/12
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:118]: and this one 14/12

File: packages/ui-library/src/theme/typography.ts [hunk 7]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
   },
-  labelLargeRegular: {
+
+  // Links
+  linkRegular: {
+    fontFamily: 'Inter',
+    fontSize: '1rem', // 16/16
+    fontStyle: 'normal',
+    fontWeight: 500,
+    lineHeight: 1.5, // 24/16
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:151]: this one should be 19.36/16

File: packages/ui-library/src/theme/typography.ts [hunk 8]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
+  },
+  linkSmall: {
     fontFamily: 'Inter',
     fontSize: '0.875rem', // 14/16
     fontStyle: 'normal',
-    fontWeight: 400,
-    lineHeight: 1.3, // 18.2/14
-    color: colors[mode].typography.typo2,
+    fontWeight: 500,
+    lineHeight: 1.5, // 21/14
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:160]: and this one 16.94/14

File: packages/ui-library/src/theme/typography.ts [hunk 9]
```diff
@@ -201,88 +86,95 @@ export const buildTypographyVariant = (mode: PaletteMode): TypographyOptions =>
...
+    color: colors[mode].accentPrimary,
   },
-  labelLargeSemibold: {
+  // Labels
+  labelLarge: {
     fontFamily: 'Inter',
     fontSize: '0.875rem', // 14/16
     fontStyle: 'normal',
     fontWeight: 600,
     lineHeight: 1.3, // 18.2/14
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:169]: 16.94/14

File: packages/ui-library/src/components/transaction-receipt/index.tsx [hunk 10]
```diff
@@ -307,7 +307,7 @@ const ERC20ApprovalTransactionReceipt = ({ transaction }: { transaction: ERC20Ap
   return (
     <>
       <StyledSectionContent>
-        <Typography variant="bodySmallLabel">
+        <Typography variant="labelRegular">
           <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="Amount approved" />
         </Typography>
         <Typography variant="bodySmallBold" sx={{ display: 'flex', alignItems: 'center', gap: spacing(2) }}>
```
Comment (FiboApe) [inline packages/ui-library/src/components/transaction-receipt/index.tsx:313]: This all should be bodyBold right?

File: packages/ui-library/src/components/transaction-receipt/index.tsx [hunk 11]
```diff
@@ -323,13 +323,13 @@ const ERC20ApprovalTransactionReceipt = ({ transaction }: { transaction: ERC20Ap
...
           <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="From address" />
         </Typography>
         <StyledBodySmallBold>{transaction.data.owner}</StyledBodySmallBold>
       </StyledSectionContent>
       <StyledSectionContent>
-        <Typography variant="bodySmallLabel">
+        <Typography variant="labelRegular">
           <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="Spender" />
         </Typography>
         <StyledBodySmallBold>{transaction.data.spender}</StyledBodySmallBold>
```
Comment (FiboApe) [inline packages/ui-library/src/components/transaction-receipt/index.tsx:335]: same as for most of StyledBodySmallBold items

File: packages/ui-library/src/components/token-picker/index.tsx [hunk 12]
```diff
@@ -465,7 +465,7 @@ const TokenPicker = ({
         </IconButton>
         <Grid container rowSpacing={5} direction="column" style={{ flexWrap: 'nowrap' }}>
           <Grid item xs={12} style={{ flexBasis: 'auto', alignSelf: 'flex-start' }}>
-            <Typography variant="h6" fontWeight={700}>
+            <Typography variant="h5Bold" color={({ palette }) => colors[palette.mode].typography.typo1}>
```
Comment (FiboApe) [inline packages/ui-library/src/components/token-picker/index.tsx:468]: double check I think this one is still used as h6

File: packages/ui-library/src/components/graph-container/index.tsx [hunk 13]
```diff
@@ -258,7 +258,7 @@ const GraphContainer = <T extends DataItem>({
       {(title || !!legend?.length) && (
         <ContainerBox justifyContent="space-between">
           {title && (
-            <Typography variant="h6Bold" color={({ palette: { mode } }) => colors[mode].typography.typo1}>
+            <Typography variant="bodyBold" color={({ palette: { mode } }) => colors[mode].typography.typo1}>
```
Comment (FiboApe) [inline packages/ui-library/src/components/graph-container/index.tsx:261]: isn't this one h5bold?

File: apps/root/src/common/components/contact-modal/contact-item/index.tsx [hunk 14]
```diff
@@ -56,8 +56,7 @@ const StyledContactItem = styled(Grid)<{ menuOpen: boolean }>`
 `}
 `;
 
-const StyledContactLabel = styled(Typography).attrs({ variant: 'h6' })`
-  font-weight: bold;
+const StyledContactLabel = styled(Typography).attrs({ variant: 'bodyBold' })`
```
Comment (FiboApe) [inline apps/root/src/common/components/contact-modal/contact-item/index.tsx:59]: Screenshot 2024-09-25 at 12 42 53 am


This should be h6 right?

File: apps/root/src/common/components/modify-settings-modal/components/changes-summary/index.tsx [hunk 15]
```diff
@@ -53,7 +53,7 @@ const ChangesSummary = ({ position, fromPrice }: ChangesSummaryProps) => {
 
   return (
     <ContainerBox flexDirection="column" gap={3} alignItems="start">
-      <Typography variant="h6" fontWeight={700}>
+      <Typography variant="bodyBold">
```
Comment (FiboApe) [inline apps/root/src/common/components/modify-settings-modal/components/changes-summary/index.tsx:56]: Screenshot 2024-09-25 at 12 44 16 am

This one should be h6

File: apps/root/src/common/components/timeline-controls/index.tsx [hunk 16]
```diff
@@ -136,7 +136,7 @@ const PositionTimelineControls = <TAction, TPosition>({
     <StyledPaper>
       <Grid container rowSpacing={6}>
         <Grid item xs={12} style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between' }}>
-          <Typography variant="h5">
+          <Typography variant="bodyBold">
```
Comment (FiboApe) [inline apps/root/src/common/components/timeline-controls/index.tsx:139]: Timeline title is h6

File: apps/root/src/common/components/token-selector/index.tsx [hunk 17]
```diff
@@ -66,7 +66,7 @@ export const TokenSelectorItem = ({
           <Typography variant="bodySmallSemibold" color={colors[mode].typography.typo2}>
             {token.name}
           </Typography>
-          <Typography variant="bodySmallLabel" color={colors[mode].typography.typo3}>
+          <Typography variant="labelRegular">
```
Comment (FiboApe) [inline apps/root/src/common/components/token-selector/index.tsx:69]: Screenshot 2024-09-25 at 12 46 30 am


Isn't this one bodySmall?

**Review by FiboApe — COMMENTED** [review]
Please review the typography variants used in the sections. I spotted the most that I could

File: packages/ui-library/src/components/transaction-receipt/index.tsx (hunk 10 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/transaction-receipt/index.tsx:313]: I'm seeing 14px in figma, so bodySmallBold seems fine

File: packages/ui-library/src/components/token-picker/index.tsx (hunk 12 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/token-picker/index.tsx:468]: It's h5Bold in figma (h6 is no longer supported in the new design system btw)

File: packages/ui-library/src/components/graph-container/index.tsx (hunk 13 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/graph-container/index.tsx:261]: Yes it is! Thanks

File: packages/ui-library/src/components/transaction-receipt/index.tsx (hunk 11 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/transaction-receipt/index.tsx:335]: Ohhh I see figma's saying Body/Bold now. But still I think thats incorrect and it should be body small (i'll ping the designers for consistency)

File: apps/root/src/common/components/contact-modal/contact-item/index.tsx (hunk 14 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/contact-modal/contact-item/index.tsx:59]: As h6 is deprecated, I based in the css props, that match with the bodyBold variant

File: apps/root/src/common/components/token-selector/index.tsx (hunk 17 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/token-selector/index.tsx:69]: Good catch sir

File: packages/ui-library/src/theme/typography.ts [hunk 18]
```diff
@@ -1,7 +1,27 @@
...
+
+const baseBodyTypography = (mode: PaletteMode) => ({
+  fontFamily: 'Inter',
+  fontSize: '1rem', // 16/16
+  fontStyle: 'normal',
+  lineHeight: 1.21, // 19.36/16
+  color: colors[mode].typography.typo2,
+});
+
+const baseBodySmallTypography = (mode: PaletteMode) => ({
```
Comment (FiboApe) [inline packages/ui-library/src/theme/typography.ts:17]: Love this!, seems we are missing the one for bodyExtraSmall

**Review by FiboApe — APPROVED** [review]
This is 🔥 🔥 🔥 🔥 🔥 🔥

**Review by FiboApe — APPROVED** [review]

**Pull request merged**